judge/runguard.c

```diff
@@ -226,6 +226,15 @@ void error(int errnum, const char *format, ...)
 {
 	va_list ap;
 	va_start(ap,format);
+	sigset_t sigs;
```

Comment:
Do you think it's better to handle this here, than simply call `terminate()` which already does signal resetting and the same sending SIG{TERM,KILL} below? (This is a genuine question.)

Reply:
I've decided to duplicate this for two reasons:
 - terminate() itself might call error(), which is unlikely, but possible. A failed kill()-call could then cause infinite recursion
 - terminate() messes with signal-handlers: It resets SIGALRM and SIGTERM to DFL. Normally, this is fine, as both are blocked anyway when terminate() is invoked during signal-handling. When calling it from the main control flow, however, there would be a short window where SIGTERM would terminate runguard while still leaving children behind, which I intended to avoid